solidity/contracts/utility/ContractRegistryClient.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity 0.4.26;
+import './Owned.sol';
+import './Utils.sol';
+import './interfaces/IContractRegistry.sol';
+
+/**
+  * @dev Base contract for ContractRegistry clients
+*/
+contract ContractRegistryClient is Owned, Utils {
+    IContractRegistry public registry;      // address of the current contract-registry
+    IContractRegistry public prevRegistry;  // address of the previous contract-registry
+    bool public allowRegistryUpdate = true; // allow/prevent changing the contract-registry
+
+    /**
+      * @dev initializes a new ContractRegistryClient instance
+      * 
+      * @param  _registry   address of a contract-registry contract
+    */
+    constructor(IContractRegistry _registry) internal validAddress(_registry) {
+        registry = IContractRegistry(_registry);
+        prevRegistry = IContractRegistry(_registry);
+    }
+
+    /**
+      * @dev returns whether or not the caller is authorized to change the contract-registry
+     */
+    function callerIsAuthorized() internal view returns (bool) {

```

Comment:
maybe call it `isCallerAdmin`? Authorized implies authorized to execute a specific action I think

Reply:
I believe that either `isAdminCaller` or `callerIsAdmin` would be more suitable.